source/s3_client.c

```diff
@@ -1515,6 +1514,12 @@ void aws_s3_client_update_connections_threaded(struct aws_s3_client *client) {
 
         struct aws_s3_request *request = aws_s3_client_dequeue_request_threaded(client);
         const uint32_t max_active_connections = aws_s3_client_get_max_active_connections(client, request->meta_request);
+        if (request->is_noop) {
+            /* If request is no-op, finishes and cleans up the request */
+            s_s3_client_meta_request_finished_request(client, request->meta_request, request, AWS_ERROR_SUCCESS);
```

Comment:
are there any lifetime issues, where maybe the meta-request completed (successfully or unsuccessfully) while this no-op request moved from the meta-req-thread to the client thread?

Reply:
nope, the no-op request keeps the meta-request around until we call `s_s3_client_meta_request_finished_request`. There is an meta-request-level track of how many request was made vs completed.